packages/react-components/src/trace-explorer/trace-explorer-time-range-data-widget.tsx

```diff
@@ -59,91 +61,65 @@ export class ReactTimeRangeDataWidget extends React.Component<
         signalManager().off(Signals.EXPERIMENT_SELECTED, this.onExperimentSelected);
         signalManager().off(Signals.EXPERIMENT_UPDATED, this.onExperimentUpdated);
         signalManager().off(Signals.EXPERIMENT_CLOSED, this.onExperimentClosed);
+        signalManager().off(Signals.CLOSE_TRACEVIEWERTAB, this.onExperimentClosed);
     };
-
+    
     private onViewRangeUpdated = (payload: TimeRangeUpdatePayload): void => {
-        const { experimentUUID: UUID, timeRange } = payload;
-
-        const update = {
-            UUID,
-            viewRange: timeRange
-        } as ExperimentTimeRangeData;
-
-        this.updateExperimentTimeRangeData(update);
+        this.experimentDataMap.updateViewRange(payload);
+        this.renderIfActive();
     };
-
+    
     private onSelectionRangeUpdated = (payload: TimeRangeUpdatePayload): void => {
-        const { experimentUUID: UUID, timeRange } = payload;
-
-        const update = {
-            UUID,
-            selectionRange: timeRange
-        } as ExperimentTimeRangeData;
-
-        this.updateExperimentTimeRangeData(update);
+        this.experimentDataMap.updateSelectionRange(payload);
+        this.renderIfActive();
+        
     };
-
+    
     private onAbsoluteRangeUpdate = (experiment: Experiment): void => {
-        if (!experiment) {
-            return;
-        }
-
-        const { UUID, start, end } = experiment;
-
-        const update = {
-            UUID,
-            absoluteRange: {
-                start,
-                end
-            }
-        } as ExperimentTimeRangeData;
-
-        this.updateExperimentTimeRangeData(update);
+        this.experimentDataMap.updateAbsoluteRange(experiment);
+        this.renderIfActive();
     };
-
-    /**
-     * Updates the data stored in experimentDataMap.  Works similar to setState() where
-     * you only input the data to change and the existing values persist.
-     * @param data Partial data of Experiment Time Range Data.
-     */
-    private updateExperimentTimeRangeData = (data: ExperimentTimeRangeData): void => {
-        const map = this.experimentDataMap;
-        const id = data.UUID;
-        const existingData = map.get(id) || {};
-        const newData = {
-            ...existingData,
-            ...data
-        };
-        map.set(id, newData);
-
-        // If the experiment is currently displayed, we need to render it
-        if (id === this.state.activeData?.UUID) {
-            this.setState({ activeData: newData });
-        }
-    };
-
-    private onExperimentUpdated = (experiment: Experiment | undefined): void => {
-        if (experiment) {
-            this.onAbsoluteRangeUpdate(experiment);
-        }
+    
+    private onExperimentUpdated = (experiment: Experiment): void => {
+        this.onAbsoluteRangeUpdate(experiment);
+        this.renderIfActive();
     };
-
+    
     private onExperimentSelected = (experiment: Experiment | undefined): void => {
```

Comment:
This changes the functionality.

if we don't have an experiement, we will set active to null. Before, it would not. I think this is desirable, but is it deliberate?

Reply:
Yes.  When the experiment is undefined it indicates an experiment has been closed.  This was deliberate.